### PR TITLE
goagen: restore goimports-style formatting

### PR DESCRIPTION
**Pull request by moorereason**
After commit f85b91bb, we lost goimports-style import path sorting.  This commit
restores that behavior and simplifies the codegen.FormatCode func.

### CODE REVIEW

**Comment by raphael** [comment]
Thank you! looks good to me. Just wondering if there was a reason @ernesto-jimenez didn't use this approach in the first place? Also please make sure to print the content of the file if `goimports` fails like the current code is doing since the file gets deleted and there's no way to know what went wrong otherwise. This code:

```go
 	if err != nil {
 -		content, _ := ioutil.ReadFile(f.Abs())
 -		var buf bytes.Buffer
 -		scanner.PrintError(&buf, err)
 -		return fmt.Errorf("%s\n========\nContent:\n%s", buf.String(), content)
 -	}
```

Thank you again!

**Comment by moorereason** [comment]
I've fixed the error message regression.  I can squash this into a single commit if you want.

**Comment by ernesto-jimenez** [comment]
@raphael @moorereason, using `goimports` can cause trouble when you have more than one package with the same names and similar signatures.

The typical example is using `template` where `imports` will import `text/template` instead of `html/template` due to its heuristics.

**Comment by ernesto-jimenez** [comment]
Note that while using `imports` there were several instances where `goa`'s generators were missing imports in the import specs and `goimports` was hiding it out, which can lead to problems later as per my previous comment where imports could import the wrong package.

Also, `goimports` can be quite slow sometimes.

**Comment by moorereason** [comment]
So, what we really want is goimports that doesn't add imports but only removes unused imports and then formats them properly.  Do we copy a bunch of code or try to get the Go team to export some `imports` funcs?

I would really like for goa's generated code to pass golint and friends.

**Comment by raphael** [comment]
That's a lot of additional code that needs to be tested and maintained just to split into stdlib vs non-stdlib imports... At the end of the day I don't know if it's worth it for generated code.... I'd be OK just using `goimports` like you had it initially. The risk as @ernesto-jimenez mentioned is that it may hide missing `import`  in generated code. I'm not too worried about getting the "wrong" import since the import shouldn't be missing in the first place and missing imports are not going to be common. The generated code always requires the same potential set of imports and that is a fairly static set.

So tl;dr we can either go back to your initial PR using goimports (and adding the printing of the source in case of failure which this latest PR didn't address) or we could decide to leave things the way they are now. Either option is OK by me - I just believe the PR in its current state is overkill.

**Comment by ernesto-jimenez** [comment]
@raphael my initial PR was because I had problems with missing imports, and there were several missing imports in the import specs. Some of which where not even picked up by the generator's tests.

I'm too worried with the size of the current code too, I'm unsure what the actual behaviour actually is.

**Comment by raphael** [comment]
yup like I commented above, happy to go either way.
